### ISSUE TITLE
Bug: False syntex error

### ISSUE THREAD

**Issue by krishnaTORQUE** [typescript-eslint/typescript-eslint#6842]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

None

### Repro Code

```typescript
const config: Record<string, string> = { backend: "http://127.0.0.1:3100" };
export default config;
```

**Error**
```shell
~/server/frontend-svelte/src/config.ts
1:50  error  Expected }  ParseError
```

### What did you expect to happen?

The error should not be there.

### What actually happened?

Its a false positive error.

### ESLint Config

```javascript
{
    "root": true,
    "env": {
        "es2021": true,
        "browser": true,
        "node": true,
        "commonjs": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/strict"
    ],
    "ignorePatterns": ["node_modules", "dist", "*.html", "*.css"],
    "overrides": [
        {
            "files": [
                "*.js",
                "*.jsx",
                "*.cjs",
                "*.mjs",
                "*.ts",
                "*.tsx",
                "*.svelte"
            ],
            "processor": "svelte3/svelte3",
            "parser": "@typescript-eslint/parser"
        }
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2021,
        "sourceType": "module",
        "parser": "@typescript-eslint/parser",
        "project": ["tsconfig.json"],
        "extraFileExtensions": [".svelte"]
    },
    "plugins": ["@typescript-eslint", "svelte3"],
    "rules": {
        "comma-dangle": "off",
        "@typescript-eslint/comma-dangle": [
            "error",
            {
                "arrays": "never",
                "objects": "always-multiline",
                "imports": "never",
                "exports": "never",
                "functions": "never"
            }
        ],
        "indent": ["error", 4],
        "linebreak-style": ["error", "unix"],
        "semi": ["error", "always"],
        "quotes": ["error", "double"],
        "quote-props": ["off", "consistent"],
        "no-await-in-loop": "error",
        "no-unreachable-loop": "error",
        "no-new-func": "error",
        "no-console": "error",
        "@typescript-eslint/array-type": "error",
        "@typescript-eslint/ban-types": "error",
        "@typescript-eslint/consistent-type-imports": "error",
        "@typescript-eslint/consistent-type-definitions": ["error", "type"],
        "@typescript-eslint/explicit-member-accessibility": "error",
        "@typescript-eslint/method-signature-style": "error",
        "@typescript-eslint/naming-convention": [
            "error",
            { "selector": "variableLike", "format": ["snake_case"] },
            { "selector": "typeLike", "format": ["PascalCase"] },
            { "selector": "method", "format": ["snake_case"] }
        ],
        "no-array-constructor": "off",
        "@typescript-eslint/no-array-constructor": "error",
        "@typescript-eslint/no-base-to-string": "error",
        "@typescript-eslint/no-confusing-non-null-assertion": "error",
        "@typescript-eslint/no-confusing-void-expression": "error",
        "no-dupe-class-members": "off",
        "@typescript-eslint/no-dupe-class-members": "error",
        "@typescript-eslint/no-duplicate-enum-values": "error",
        "@typescript-eslint/no-duplicate-type-constituents": "error",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-extra-non-null-assertion": "error",
        "no-extra-semi": "off",
        "@typescript-eslint/no-extra-semi": "error",
        "@typescript-eslint/no-floating-promises": "off",
        "no-implied-eval": "off",
        "@typescript-eslint/no-implied-eval": "error",
        "@typescript-eslint/no-inferrable-types": "off",
        "@typescript-eslint/no-invalid-void-type": "error",
        "@typescript-eslint/no-meaningless-void-operator": "error",
        "@typescript-eslint/no-misused-new": "error",
        "@typescript-eslint/no-misused-promises": "error",
        "@typescript-eslint/no-mixed-enums": "error",
        "@typescript-eslint/no-namespace": "error",
        "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
        "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
        "@typescript-eslint/no-non-null-assertion": "error",
        "@typescript-eslint/no-this-alias": "error",
        "no-throw-literal": "off",
        "@typescript-eslint/no-throw-literal": "error",
        "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
        "@typescript-eslint/no-unnecessary-condition": "error",
        "@typescript-eslint/no-unnecessary-type-arguments": "error",
        "@typescript-eslint/no-unnecessary-type-constraint": "error",
        "@typescript-eslint/no-unsafe-declaration-merging": "error",
        "@typescript-eslint/no-unsafe-return": "error",
        "no-unused-vars": "off",
        "@typescript-eslint/no-unused-vars": "error",
        "no-use-before-define": "off",
        "@typescript-eslint/no-use-before-define": "error",
        "no-useless-constructor": "off",
        "@typescript-eslint/no-useless-constructor": "error",
        "@typescript-eslint/no-useless-empty-export": "error",
        "@typescript-eslint/non-nullable-type-assertion-style": "error",
        "@typescript-eslint/prefer-string-starts-ends-with": "error",
        "@typescript-eslint/promise-function-async": "error",
        "@typescript-eslint/switch-exhaustiveness-check": "error"
    },
    "settings": {
        "svelte3/typescript": true
    }
}
```

### tsconfig

```jsonc
{
    "extends": "@tsconfig/svelte/tsconfig.json",
    "compilerOptions": {
        "target": "ESNext",
        "useDefineForClassFields": true,
        "module": "ESNext",
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "allowJs": true,
        "checkJs": true
    },
    "include": [
        "src/**/*.d.ts",
        "src/**/*.ts",
        "src/**/*.js",
        "src/**/*.svelte"
    ],
    "references": [{ "path": "./tsconfig.node.json" }]
}
```

### Expected Result

The error should not be there.

### Actual Result

Its a false positive error.

**Comment by bradzacher** [comment]
> Playground Link
> None 

Please don't ignore the issue template. 

The reason we ask for playground links for bug reports is because it is an isolated environment within which to reproduce the bug.

If you can't reproduce it in the playground then it often means it's not a bug in our tooling.

[I dumped your code into the playground and got no error](https://typescript-eslint.io/play/#ts=5.0.2&sourceType=module&code=MYewdgzgLgBKYDMCWBzAXDASgU1AJwBMAeaPJMFAGhlPJQD4YBeGAbxgCMBDYAa2zAEMAIgAWUKAAc0AehkBGAEwB2AHQAGDavloAzPPXrhMAL4BuAFDYAHpJB5YBbAi4BXADax4yFGaA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA) - so it's not our parser throwing the error - it's something else.

You'll need to take some time narrowing down the problem before we can help out here. Glancing at your config - I see `"processor": "svelte3/svelte3"` is turned on for TS files. If I had to guess - this is likely something to do with the problem.

I'm going to close this, but if you figure out an isolated reproduction - comment back and happy to reopen.
